manifests/nodes.pp

```diff
@@ -55,13 +55,13 @@
     package { $system_packages: ensure => installed }
   }
 
-  $python_packages = hiera_array( 'python_packages', [] )
+  $python_packages = hiera_hash( 'python_packages', {} )
   if !empty($python_packages) {
-    package { $python_packages:
-      ensure   => installed,
-      provider => 'pip',
```

Comment:
We need to put the provider / require in the defaults, otherwise it tries to install via apt, not pip